symfem/elements/bernardi_raugel.py

```diff
@@ -87,34 +92,22 @@ def __init__(self, reference: Reference, order: int):
                 v1 = reference.vertices[edge[0]]
                 v2 = reference.vertices[edge[1]]
                 midpoint = tuple(sympy.Rational(i + j, 2) for i, j in zip(v1, v2))
-                d = tuple(j - i for i, j in zip(v1, v2))
-                dofs.append(
-                    DotPointEvaluation(
-                        reference, midpoint, d, entity=(1, e_n), mapping="contravariant"
-                    )
-                )
-            for f_n in range(reference.sub_entity_count(2)):
-                face = reference.sub_entity(2, f_n)
-                normal = tuple(i * face.jacobian() for i in face.normal())
-                for e_n in range(3):
-                    edge_entity = face.sub_entity(1, e_n)
-                    midpoint = tuple(
-                        sympy.Rational(i + j, 2) for i, j in zip(*edge_entity.vertices)
-                    )
+                for i in range(tdim):
+                    direction = tuple(1 if i == j else 0 for j in range(tdim))
                     dofs.append(
                         DotPointEvaluation(
-                            reference, midpoint, normal, entity=(2, f_n), mapping="contravariant"
+                            reference, midpoint, direction, entity=(1, e_n), mapping="identity"
                         )
                     )
 
             p = Lagrange(reference, 0, variant="equispaced")
-
             for i in range(3):
                 dofs.append(
                     DivergenceIntegralMoment(
-                        reference, x[i], p.dofs[0], entity=(3, 0), mapping="contravariant"
+                        reference, x[i], p.dofs[0], entity=(3, 0), mapping="identity"
```

Comment:
I am not sure why you have divergence moments, but the mapping for this dofs should contravariant. The canonical face DOFs are normal moments, and should map contravariantly.

Reply:
I need to look at these again. I've opened #283

Reply:
Looking at this element again. From the Bernardi-Raugel paper, it looks like these are right:

![image](https://github.com/user-attachments/assets/4bebd09e-19fb-4c5d-980f-bcc341d2834c)